core/src/main/java/org/jboss/pnc/rex/core/QueueManagerImpl.java

```diff
@@ -44,111 +47,151 @@ public class QueueManagerImpl implements QueueManager {
     private final Counter running;
     private final TaskRegistry container;
     private final TaskController controller;
+    private final FaultToleranceDecorator ft;
 
-    public QueueManagerImpl(@MaxConcurrent Counter max, @Running Counter running, TaskRegistry container, TaskController controller) {
+    public QueueManagerImpl(@MaxConcurrent Counter max,
+                            @Running Counter running,
+                            TaskRegistry container,
+                            TaskController controller,
+                            FaultToleranceDecorator ft) {
         this.max = max;
         this.running = running;
         this.container = container;
         this.controller = controller;
+        this.ft = ft;
     }
 
     @Override
     @Transactional
     public void poke() {
-        log.info("QUEUE: Poking Task queue");
-        VersionedValue<Long> maxMetadata = max.getMetadataValue();
-        Long maxValue = maxMetadata.getValue();
-
-        VersionedValue<Long> runningMetadata = running.getMetadataValue();
-        Long runningValue = runningMetadata.getValue();
-
-        if (runningValue >= maxValue) {
-            log.debug("QUEUE: Maximum number of parallel builds reached.({} out of {})", runningValue, maxValue);
-            return;
+        log.info("QUEUE: Poking Task queues");
+        Map<String, Long> maxEntries = max.entries();
+        Map<String, Long> runningEntries = running.entries();
+
+        Set<String> consideredQueues = new HashSet<>();
+        for (var queue : maxEntries.keySet()) {
+            Long maxValue = maxEntries.get(queue);
+            Long runningValue = runningEntries.get(queue);
+
+            if (runningValue >= maxValue) {
+                log.debug("QUEUE '{}': Maximum number of parallel builds reached.({} out of {})",
```

Comment:
`tasks` not `builds` at this point I guess right?

Reply:
Oh... yeah. Good point.